### DISCUSSION TITLE
Prisma updateAt column with Microsoft sql server

### DISCUSSION THREAD

**Discussion by lukasz0707** [prisma/prisma#22401] [Q&A]
### Question

Is there a way for updatedAt column to insert current date on change. Right now in schema.prisma I am using `updatedAt DateTime @default(now()) @updatedAt.` But it does not work.

**Comment by ludralph** [comment]
Hi @lukasz0707 

Thank you for raising this question.

There is an open [issue](https://github.com/prisma/prisma/issues/9317) that describes this exact situation you are experiencing and a workaround for now is to customize your migration as stated in that issue.
> You can [customize your migration](https://www.prisma.io/docs/concepts/components/prisma-migrate/migrate-development->production#customizing-migrations) and include the native SQL command yourself for the "updated at"-logic using the `--create-only` option, e.g.:
>
>```
>npx prisma migrate dev --create-only
>```
>
>Then inside the SQL, add the respective command for your DB, e.g. for MySQL:
>
>```sql
>ALTER TABLE mytable
>ADD COLUMN updated_at DATETIME DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP;
>```
>
>Afterwards, run `prisma migrate dev` again to apply the migration 🙌 

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by lukasz0707** [reply]
I think that would work for MySQL but it does not work for Microsoft SQL Server. I created a trigger directly inside the database and it works. But I want to know if I can somehow include that in the migration.

**Reply by ludralph** [reply]
Hi @lukasz0707,  

Yes, you can declare database triggers that are a part of migrations. You can create a migration with `--create-only` flag. So invoking the command `npx prisma migrate dev --create-only` will create an empty migration file. You can then add the database trigger in this newly created file. Invoking `npx prisma migrate dev` again would then apply this new migration.

You can read more about customizing migrations [here](https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/customizing-migrations).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
